app/rpc/namespaces/eth/api.go

```diff
@@ -1015,10 +1016,17 @@ func (api *PublicEthereumAPI) EstimateGas(args rpctypes.CallArgs) (hexutil.Uint6
 		return 0, TransformDataError(err, "eth_estimateGas")
 	}
 
-	// TODO: change 1000 buffer for more accurate buffer (eg: SDK's gasAdjusted)
 	estimatedGas := simResponse.GasInfo.GasUsed
+	if estimatedGas > evmtypes.DefaultMaxGasLimitPerTx {
```

Comment:
use evmKeeper.GetParams(ctx).MaxGasLimitPerTx instead of DefaultMaxGasLimitPerTx, because of MaxGasLimitPerTx could be changed by gov params.

Reply:
put the estimate gas logic to the handler.use path to distinguish eth_estimateGas eth_call.